library/core/tests/waker.rs

```diff
@@ -20,3 +22,26 @@ static WAKER_VTABLE: RawWakerVTable = RawWakerVTable::new(
     |_| {},
     |_| {},
 );
+
+const NOP_RAWWAKER: RawWaker = {
+    fn nop(_: *const ()) {}
+    const VTAB: RawWakerVTable = RawWakerVTable::new(|_| NOP_RAWWAKER, nop, nop, nop);
+    RawWaker::new(&() as *const (), &VTAB)
+};
+
+const NOP_WAKER: &Waker = &unsafe { Waker::from_raw(NOP_RAWWAKER) };
+
+const NOP_CONTEXT: Context<'static> = Context::from_waker(NOP_WAKER);
+
+fn poll_once<T, F>(f: &mut F) -> Poll<T>
+where
+    F: Future<Output = T> + ?Sized + Unpin,
+{
+    let mut cx = NOP_CONTEXT;
+    Pin::new(f).as_mut().poll(&mut cx)
+}
+
+#[test]
+fn test_const_waker() {
+    assert_eq!(poll_once(&mut ready(1)), Poll::Ready(1));
+}
```

Comment:
Could you put this into a module so the related things are self-contained, and add a comment linking https://github.com/rust-lang/rust/issues/102012#issuecomment-1915282956?